### PR TITLE
Response to comments

### PR DESCRIPTION
**Pull request by wwwtyro**
Here you go, @ibgreen. Wow, git really does not like `$$` in the comment string. Commit `d243893`'s comment is `nix $$family`.

@philogb

### CODE REVIEW

File: src/math/array-impl.js [hunk 1]
```diff
@@ -214,7 +210,7 @@ var generics = {
   },
 
   clone(dest) {
-    if (dest.$$family) {
+    if (dest.constructor.name == 'Vec3') {
       return new Vec3(dest[0], dest[1], dest[2]);
```
Comment (ibgreen) [inline src/math/array-impl.js]: I'd really prefer

   if (dest instanceof Vec3)

Would that work?

File: src/scene.js [hunk 2]
```diff
@@ -186,8 +180,8 @@ export default class Scene {
   render(opt = {}) {
     const camera = this.camera;
     const {renderProgram} = opt;
-    const pType = type(this.program);
-    const multiplePrograms = !renderProgram && pType === 'object';
+    const pType = this.program.constructor.name;
```
Comment (ibgreen) [inline src/scene.js]: Is pType actually used? Can we drop?

File: src/utils.js [hunk 3]
```diff
@@ -15,48 +20,44 @@ export function merge() {
...
-  if (t == 'object') {
+  var t = elem.constructor.name, ans;
+  if (t == 'Object') {
     ans = {};
     for (var p in elem) {
       ans[p] = detach(elem[p]);
     }
     return ans;
-  } else if (t == 'array') {
+  } else if (t == 'Array') {
```
Comment (ibgreen) [inline src/utils.js]: Use Array.isArray

File: src/utils.js [hunk 4]
```diff
@@ -15,48 +20,44 @@ export function merge() {
...
+/**
+* Internal function for duplicating an object.
+* @param {object} elem - The object to recursively duplicate.
+* @returns {object}
+**/
 function detach(elem) {
-  var t = type(elem), ans;
-  if (t == 'object') {
+  var t = elem.constructor.name, ans;
+  if (t == 'Object') {
```
Comment (ibgreen) [inline src/utils.js]: typeof t === 'object' && t !== null

File: test/unit/testcore.js [hunk 5]
```diff
@@ -1,9 +0,0 @@
-import {PhiloGL} from '../../src';
-import test from 'tape';
```
Comment (ibgreen) [inline test/unit/testcore.js]: I think a few basic test cases for the Utils would be very quick to write and well worth it

**Comment by ibgreen** [comment]
Thanks, looks great. Just a few nits...

**Comment by wwwtyro** [comment]
@ibgreen Thanks for reviewing! I fixed a couple of things according to your comments. `t` in utils/detach is a string representing the type, so not sure how to move forward there. I'll make the tests right now!

**Comment by wwwtyro** [comment]
@ibgreen tests up :)

**Pull request merged**